src/test/kotlin/org/rust/lang/refactoring/RsExtractFunctionTest.kt

```diff
@@ -726,6 +726,76 @@ class RsExtractFunctionTest : RsTestBase() {
         false,
         "bar")
 
+        fun `test extract a function with reference parameter`() = doTest("""
```

Comment:
Let's split this test into several small tests to check one case per tests
I think it's more convenient to check what cases are tested